public/app/features/dashboard-scene/panel-edit/PanelEditorRenderer.tsx

```diff
@@ -57,7 +59,9 @@ export function PanelEditorRenderer({ model }: SceneComponentProps<PanelEditor>)
 
 function VizAndDataPane({ model }: SceneComponentProps<PanelEditor>) {
   const dashboard = getDashboardSceneFor(model);
-  const { vizManager, dataPane } = model.useState();
+  const { vizManager, dataPane, showLibraryPanelSaveModal } = model.useState();
+  const { sourcePanel } = vizManager.useState();
+  const libraryPanel = sourcePanel.resolve().parent;
```

Comment:
```suggestion
  const libraryPanel = getLibraryPanel(sourcePanel.resolve());
```
Looking at this change, I wonder if it is worth to have the util getLibraryPanel. In the end, is the parent, and we're using the parent property for some other cases like GridItem. I'd remove the util, wdyt?

Reply:
I changed it to your suggestion. I like the util because it removes a lot of `instanceof LibraryVizPanel` in the code. Using the util here as you suggested made it possible to just check if library panel is not undefined on line 88